File: AlgorithmStudy/10주차/BOJ7795/김재환7795.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int T = Integer.parseInt(br.readLine());
        while (T-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int N = Integer.parseInt(st.nextToken());
            int M = Integer.parseInt(st.nextToken());

            int[] A = new int[N];
            int[] B = new int[M];

            st = new StringTokenizer(br.readLine());
            for (int i = 0; i < N; i++) {
                A[i] = Integer.parseInt(st.nextToken());
            }

            st = new StringTokenizer(br.readLine());
            for (int i = 0; i < M; i++) {
                B[i] = Integer.parseInt(st.nextToken());
            }

            // B 정렬
            Arrays.sort(B);

            long ans = 0;
            for (int a : A) {
                // 이분탐색으로 a보다 작은 B 개수를 찾는다
                int left = 0;
                int right = M;
                while (left < right) {
                    int mid = (left + right) / 2;
                    if (B[mid] < a) left = mid + 1;
                    else right = mid;
                }
                ans += left; // left는 B에서 a보다 작은 원소 개수
            }
            sb.append(ans).append('\n');
        }
        System.out.print(sb.toString());
    }
}

File: AlgorithmStudy/10주차/BOJ7795/김재환7795.py
import sys
import bisect
input = sys.stdin.readline

T = int(input())

for _ in range(T):
    N, M = map(int, input().split())
    A = list(map(int, input().split()))
    B = list(map(int, input().split()))
    
    # B는 정렬해두기
    B.sort()
    
    ans = 0
    for a in A:
        # bisect_left(B, a)는 B에서 a보다 작은 수들의 개수
        # a보다 작은 값이 모두 왼쪽에 오므로 인덱스가 개수
        ans += bisect.bisect_left(B, a)
    
    print(ans)
